File: 0532 k-diff-pairs-in-an-array/solution.java
class Solution {
    public int findPairs(int[] nums, int k) {
      Set<Integer> s = new HashSet<>();
      Set<Integer> s1 = new HashSet<>();

      for(int i:nums){
        if(s.contains(i-k)) s1.add(i-k);
        if(s.contains(i+k)) s1.add(i);
        s.add(i);
      }
      return s1.size();
    }
}

File: 0532 k-diff-pairs-in-an-array/solution.go
func findPairs(nums []int, k int) int {
    s1:=make(map[int]bool)
    s2:=make(map[int]bool)

    for _,v:=range nums{
        if _,ok := s1[v-k]; ok {
            s2[v-k]=true
        } 
        if _,ok :=s1[v+k]; ok {
            s2[v]=true
        } 
        s1[v]=true
    }
    return len(s2)
}
